[4.0] com_menu - Change Featured Icon to Home Icon in

coolcat-creations opened the pull request:
Pull Request for Issue  #28976 

### Summary of Changes
Changed the Default Icon in Menus to a home icon. If it has to be called home or default could be discussed in another issue. This PR makes a difference between the _featured - action_ and the _set menu item as default_ action. It also adds the possibility to pass the active and inactive iconclass suffix to the isDefault method.


### Testing Instructions
apply patch 
see if the home Icon on monolingual sites is now a home icon
the non-default items are for now a simple circle
on a multilingual website the home icon should be still a flag like before.

### Expected result

<img width="557" alt="grafik" src="https://user-images.githubusercontent.com/828371/81486584-f516a980-9255-11ea-988f-4c6fce1888be.png">

Sidebar:
<img width="317" alt="grafik" src="https://user-images.githubusercontent.com/828371/81486624-53dc2300-9256-11ea-96d8-7f0c975cfe56.png">




### Actual result
<img width="549" alt="grafik" src="https://user-images.githubusercontent.com/828371/81486298-16769600-9254-11ea-9ae7-db674da5feee.png">

coolcat-creations committed:
Change Featured Icon to Home Icon in com_menu

coolcat-creations committed:
Change Icon class

coolcat-creations committed:
Change Default icon in sidebar menu too

infograf768 commented:
Remark 1:

Size of icon is very different between sidemenu and Menu dasboard. Menu dashboard is better imho.
<img width="1247" alt="Screen Shot 2020-05-10 at 07 37 08" src="https://user-images.githubusercontent.com/869724/81491778-417fda80-9292-11ea-9393-cdaec3d5eb1e.png">

Remark 2:
Density of Yellow is very different between the featured star (for articles manager) and Home icon set to All languages in menu items manager. Articles manager star density is better imho.

<img width="348" alt="Screen Shot 2020-05-10 at 07 41 57" src="https://user-images.githubusercontent.com/869724/81491800-8c99ed80-9292-11ea-8e51-b037be709e46.png">
<img width="333" alt="Screen Shot 2020-05-10 at 07 43 33" src="https://user-images.githubusercontent.com/869724/81491803-93286500-9292-11ea-8265-9675f9df562e.png">

The same colour is applied
```
color: var(--warning);
border-color: var(--warning);
```
The issue comes from 
```
.disabled {
    opacity: .4;
}
```

coolcat-creations commented:
Yes, this PR is only about to change the type of the icon, not the color or size - for color we would need another class in the table header and for sizes it’s also not bound to only the home icon but the general interface.

adj9 commented:
I have tested this item :white_check_mark: successfully on 109a7b20433d7d1820777540d36def151723d8fd<br /><br />Apply the PR and I've change the icon for menu.
I've do an article in person lang and see the icon should<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29016">issues.joomla.org/tracker/joomla-cms/29016</a>.</sub>

coolcat-creations commented:
> I have tested this item ✅ successfully on [109a7b2](https://github.com/joomla/joomla-cms/commit/109a7b20433d7d1820777540d36def151723d8fd)
> 
> Apply the PR and I've change the icon for menu.
> I've do an article in person lang and see the icon should
> ![image](https://user-images.githubusercontent.com/5918444/81496081-3a1df880-92b5-11ea-952f-38f136da786e.png)
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/29016](https://issues.joomla.org/tracker/joomla-cms/29016).

Thank you!

coolcat-creations committed:
Different class for disabled Home item

coolcat-creations commented:
@adj9 @infograf768  I added the styling for the home item to not have the disabled class and added a "not-allowed" cursor. Can you retest? 
For the icon sizes I think a separate PR would be better.

infograf768 commented:
will do tomorrow.

infograf768 commented:
I have tested this item :white_check_mark: successfully on 38bc0209e37cdbfab870021933580e686de449bd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29016">issues.joomla.org/tracker/joomla-cms/29016</a>.</sub>

Quy commented on administrator/components/com_menus/tmpl/items/default.php:
```diff
@@ -209,7 +209,7 @@
 										<?php if ($item->type == 'component') : ?>
 											<?php if ($item->language == '*' || $item->home == '0') : ?>
-												<?php echo HTMLHelper::_('jgrid.isdefault', $item->home, $i, 'items.', ($item->language != '*' || !$item->home) && $canChange && !$item->protected); ?>
+												<?php echo HTMLHelper::_('jgrid.isdefault', $item->home, $i, 'items.', ($item->language != '*' || !$item->home) && $canChange && !$item->protected, "cb", null, "home", "circle"); ?>
```
```suggestion
												<?php echo HTMLHelper::_('jgrid.isdefault', $item->home, $i, 'items.', ($item->language != '*' || !$item->home) && $canChange && !$item->protected, 'cb', null, 'home', 'circle'); ?>
```

infograf768 replied:
Agree. Would not change test OK.

infograf768 commented:
@coolcat-creations 

I suggest you also add a few lines above
```
  &.featured-disabled {
    opacity: 1;
  }
```
as this will now also solve density in Installed Languages and Template Styles for the default star colour.

N6REJ commented:
@coolcat-creations 
```
  &.disabled .icon-home {
    color: $state-warning-bg;
    border-color: $state-warning-bg;
  }
```
should be 
```
&.disabled fas fa-home,
  &.disabled .icon-home {
    color: $state-warning-bg;
    border-color: $state-warning-bg;
  }
```

```
  &.disabled fas fa-home {
    color: $state-warning-bg;
    border-color: $state-warning-bg;
  }
```
is preferred, but i understand for b/c that may not be possible?

coolcat-creations committed:
Update administrator/components/com_menus/tmpl/items/default.php

Co-authored-by: Quy <quy@fluxbb.org>

coolcat-creations commented:
@N6REJ - I can't add code that is not applying yet, you would need to add this into your PR after mine is merged or me after yours is merged, but not before both are merged.

coolcat-creations commented:
> &.featured-disabled {
>     opacity: 1;
>   }

May I create a seperate PR for that? I prefer not to mix up too much in case something needs to be looked up later or reverted.

infograf768 commented:
> @infograf768 - May I create a seperate PR for that? I prefer not to mix up too much in case something needs to be looked up later or reverted.

Sure, but don’t forget! 😺

infograf768 commented:
I have tested this item :white_check_mark: successfully on 0838c6daff3984d89e9ef096b2077b0261d6415c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29016">issues.joomla.org/tracker/joomla-cms/29016</a>.</sub>

Quy commented:
I have tested this item :white_check_mark: successfully on 0838c6daff3984d89e9ef096b2077b0261d6415c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29016">issues.joomla.org/tracker/joomla-cms/29016</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29016">issues.joomla.org/tracker/joomla-cms/29016</a>.</sub>

infograf768 commented:
Tks

infograf768 commented:
@coolcat-creations 
Now you can make the PR for https://github.com/joomla/joomla-cms/pull/29016#issuecomment-626482419 ;)
